### PR TITLE
merge wallet, shelley wallet

### PR DESCRIPTION
**Pull request by wolverineks**

### CODE REVIEW

File: src/yoroi-wallets/cardano/ShelleyWallet.ts [hunk 1]
```diff
@@ -7,8 +7,10 @@ import cryptoRandomString from 'crypto-random-string'
 import ExtendableError from 'es6-error'
```
Comment (banklesss) [inline src/yoroi-wallets/cardano/ShelleyWallet.ts:7]: should this file be in `src/yoroi-wallets/cardano/shelley`?

Comment (wolverineks) [reply src/yoroi-wallets/cardano/ShelleyWallet.ts:7]: probably be renamed to CardanoWallet, its not really just shelley though

File: src/yoroi-wallets/cardano/ShelleyWallet.ts [hunk 2]
```diff
@@ -863,6 +914,285 @@ export class ShelleyWallet extends Wallet implements WalletInterface {
...
+
+  state: WalletState = {
+    lastGeneratedAddressIndex: 0,
+  }
+
+  isInitialized = false
+
+  transactionCache: null | TransactionCache = null
+
+  _doFullSyncMutex: any = {name: 'doFullSyncMutex', lock: null}
```
Comment (banklesss) [inline src/yoroi-wallets/cardano/ShelleyWallet.ts:926]: private(s)?

File: src/Settings/ChangePassword/ChangePasswordScreen.tsx [hunk 3]
```diff
@@ -167,9 +167,8 @@ const useChangePassword = (
   wallet: YoroiWallet,
   mutationOptions: MutationOptions<void, Error, {currentPassword: string; newPassword: string}>,
 ) => {
-  const intl = useIntl()
   const {mutate, ...mutation} = useMutation(
-    ({currentPassword, newPassword}) => wallet.changePassword(currentPassword, newPassword, intl),
+    ({currentPassword, newPassword}) => wallet.changePassword(currentPassword, newPassword),
```
Comment (stackchain) [inline src/Settings/ChangePassword/ChangePasswordScreen.tsx:171]: hmm, missed the update on the interface in my PR, ty @wolverineks

File: src/yoroi-wallets/cardano/types.ts [hunk 4]
```diff
@@ -250,17 +261,23 @@ export type SignedTxLegacy = {
...
+  sync: () => Promise<void>
+  resync: () => Promise<void>
+  signTx(unsignedTx: YoroiUnsignedTx, rootKey: string): Promise<YoroiSignedTx>
+  submitTransaction: (signedTx: string) => Promise<[]>
+  subscribe: (subscription: WalletSubscription) => Unsubscribe
+
+  // NonNullable
   networkId: NonNullable<WalletInterface['networkId']>
   walletImplementationId: NonNullable<WalletInterface['walletImplementationId']>
   defaultAsset: DefaultAsset
```
Comment (stackchain) [inline src/yoroi-wallets/cardano/types.ts:277]: following this logic, `networkId` and `defaultAsset` can be moved upwards, but I'm guessing `initialize()` is about to die.

**Review by stackchain — APPROVED** [review]

File: src/yoroi-wallets/cardano/types.ts (hunk 4 above)
Comment (wolverineks) [reply src/yoroi-wallets/cardano/types.ts:277]: eventually,
`await Wallet.create`
`await Wallet.createBip44`
same as transaction cache

Comment (wolverineks) [reply src/yoroi-wallets/cardano/types.ts:277]: this will eliminate maybe properties

**Pull request merged**